app/org/sagebionetworks/bridge/models/schedules/TaskScheduler.java

```diff
@@ -51,43 +41,46 @@ protected DateTime getScheduledTimeBasedOnEvent(Schedule schedule, Map<String, D
         return eventTime;
     }
     
-    protected void addTaskForEachTime(List<Task> tasks, DateTime scheduledTime) {
+    protected void addTaskForEachTime(List<Task> tasks, ScheduleContext context, DateTime scheduledTime) {
         if (schedule.getTimes().isEmpty()) {
             // We're using whatever hour/minute/seconds were in the original event.
-            addTaskForEachActivityAtTime(tasks, scheduledTime);
+            addTaskForEachActivityAtTime(tasks, context, scheduledTime);
         } else {
             for (LocalTime time : schedule.getTimes()) {
                 scheduledTime = new DateTime(scheduledTime).withTime(time);
-                addTaskForEachActivityAtTime(tasks, scheduledTime);
+                addTaskForEachActivityAtTime(tasks, context, scheduledTime);
             }
         }
     }
     
-    private void addTaskForEachActivityAtTime(List<Task> tasks, DateTime scheduledTime) {
+    private void addTaskForEachActivityAtTime(List<Task> tasks, ScheduleContext context, DateTime scheduledTime) {
         // If this time point is outside of the schedule's active window, skip it.
         if (isInWindow(schedule, scheduledTime)) {
             // As long at the tasks are not already expired, add them.
-            DateTime expiresOn = getExpiresOn(scheduledTime, schedule);
-            if (expiresOn == null || expiresOn.isAfter(now)) {
+            DateTime expiresOn = getExpiresOn(schedule, scheduledTime);
+            if (expiresOn == null || expiresOn.isAfter(context.getNow())) {
```

Comment:
Intervals only go down to hours so by and large, yes, but still, I've changed the context to capture now on construction and use that for the duration of the request.